be/src/olap/olap_server.cpp

```diff
@@ -523,11 +523,17 @@ std::vector<TabletSharedPtr> StorageEngine::_generate_compaction_tasks(
                             ? copied_cumu_map[data_dir]
                             : copied_base_map[data_dir],
                     &disk_max_score, _cumulative_compaction_policy);
-            if (tablet != nullptr) {
+            if (tablet != nullptr &&
+                !tablet->tablet_meta()->mutable_tablet_schema()->disable_auto_compaction()) {
```

Comment:
inline const TabletSchema& TabletMeta::tablet_schema() const {
    return *_schema;
}

inline TabletSchema* TabletMeta::mutable_tablet_schema() {
    return _schema.get();
}

why use mutable_tablet_schema() not tablet_schema()?